src/server/scripts/Events/hallows_end.cpp

```diff
@@ -0,0 +1,534 @@
+            {
+                case 87: // Goldshire
+                    x = -9494.4f;
+                    y = 48.53f;
+                    z = 70.5f;
+                    o = 0.5f;
+                    path = 235431;
+                    break;
+                case 131: // Kharanos
+                    x = -5558.34f;
+                    y = -499.46f;
+                    z = 414.12f;
+                    o = 2.08f;
+                    path = 235432;
+                    break;
+                case 3576: // Azure Watch
+                    x = -4163.58f;
+                    y = -12460.30f;
+                    z = 63.02f;
+                    o = 4.31f;
+                    path = 235433;
+                    break;
+                case 362: // Razor Hill
+                    x = 373.2f;
+                    y = -4723.4f;
+                    z = 31.2f;
+                    o = 3.2f;
+                    path = 235434;
+                    break;
+                case 159: // Brill
+                    x = 2195.2f;
+                    y = 264.0f;
+                    z = 55.62f;
+                    o = 0.15f;
+                    path = 235435;
+                    break;
+                case 3665: // Falconwing Square
+                    x = 9547.91f;
+                    y = -6809.9f;
+                    z = 27.96f;
+                    o = 3.4f;
+                    path = 235436;
+                    break;
+                default:
+                    x = 0.0f;
+                    y = 0.0f;
+                    z = 0.0f;
+                    path = 0;
+                    break;
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_BEGIN:
+                    {
+                        float x, y, z, o;
+                        uint32 path;
+                        GetInitXYZ(x, y, z, o, path);
+                        if (Creature* shadeOfHorseman = me->SummonCreature(NPC_SHADE_OF_HORSEMAN, x, y, z, o, TEMPSUMMON_CORPSE_TIMED_DESPAWN, 10000))
+                        {
+                            shadeOfHorseman->GetMotionMaster()->MovePath(path, true);
+                            shadeOfHorseman->AI()->DoAction(path);
+                        }
+                        _events.Repeat(Minutes(40));
+                        break;
+                    }
+                }
+            }
+        }
+
+    private:
+        EventMap _events;
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_costumed_orphan_matronAI(creature);
+    }
+};
+
+class npc_soh_fire_trigger : public CreatureScript
+{
+public:
+    npc_soh_fire_trigger() : CreatureScript("npc_soh_fire_trigger") { }
+
+    struct npc_soh_fire_triggerAI : public NullCreatureAI
```

Comment:
```
src/server/scripts/Events/hallows_end.cpp:180:44: fatal error:
 unknown class name 'NullCreatureAI'; did you mean 'CreatureAI'?

    struct npc_soh_fire_triggerAI : public NullCreatureAI
                                           ^~~~~~~~~~~~~~
                                           CreatureAI

src/server/game/Entities/Creature/Creature.h:31:7
Building CXX object src/server/scripts/CMakeFiles/scripts_commands.dir/Commands/cs_ahbot.cpp.o:
 note: 'CreatureAI' declared here
class CreatureAI;
      ^

1 error generated.
```

Reply:
Why wasn't this error triggered before? Recent changes?

Reply:
#include "PassiveAI.h"